### ISSUE TITLE
Type mismatch for int function parameter that is used in a condition

### ISSUE THREAD

**Issue by MartinSpiessl** [verifast/verifast#357]
When I try to verify the following annotated C program with verifast, I get `Type mismatch. Actual: bool. Expected: int`:

main.c:
```C
int diff(int a, int b, int fallback)
//@ requires true;
//@ ensures result >= 0 || result == fallback;
{
  if (fallback) return fallback;
  if (b>a) return b-a;
    return a-b;
}

int main()
//@ requires true;
//@ ensures true;
{
  int a = 5, b = 7;
  int ret = diff(a,b,a>b);
  return ret;
}
```

command line:
```
bin/verifast -disable_overflow_check main.c
main.c
main.c(15,23-24): Type mismatch. Actual: bool. Expected: int.
```

I used the latest verifast release version on Linux: https://github.com/verifast/verifast/releases/tag/21.04

The problem seems to be that the parameter `fallback` of the function `diff` is declared as type `int`, but used as condition in the `if`-statement. This cannot be fixed by changing the type of the parameter to `bool`, since then we get the same type mismatch at other positions where `fallback` is used as an `int`.

`main.c` is a perfectly fine C program, so the verification should not require any changes to the program itself.

**Comment by btj** [comment]
No, the problem here is that VeriFast does not support using expression `a>b`, which it sees as being of type `bool`, in a place where an `int` value is expected. Writing `a>b ? 1 : 0` would work. I'll see if I can fix it easily.
